src/runtime_src/core/common/api/xrt_kernel.cpp

```diff
@@ -3171,6 +3199,28 @@ get_regmap_size(const xrt::kernel& kernel)
     return kernel.get_handle()->get_regmap_size();
 }
 
+std::string
+get_kernel_name(const xrt::kernel& kernel)
+{
+  return kernel.get_handle()->get_name();
+}
+
+xrt::hw_context
+get_hw_ctx(const xrt::kernel& kernel)
+{
+  return kernel.get_handle()->get_hw_context();
+}
+
+xrt::kernel
+create_kernel_from_implementation(void* kernel_impl)
```

Comment:
```suggestion
create_kernel_from_implementation(const hwctx_impl* kernel_impl)
```
At least use the hwctx_impl pointer.  I am not sure about the `const` in context of getting shared from this and `const` would require the shared_ptr<base_logger> to be mutable for sure.